pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -106,22 +106,22 @@ let
   crossCabalFlagsString =
     stdenv.lib.optionalString isCross (" " + stdenv.lib.concatStringsSep " " crossCabalFlags);
 
-  defaultConfigureFlags = [
+  defaultConfigureFlags = flatten [
```

Comment:
It is a refactoring unrelated to the change, so I'm going to take it out.

Just for elaboration: The motivation (as mentioned in the PR) is that if one adds a flag to `defaultConfigureFlags` that is conditional on `isGhcjs` with `optionalString`, a reasonable expectation, IMHO, would be that only GHCJS package derivations change and have to be rebuilt. This is not the case, since the flags are joined by `concatStringsSep " "`, every empty string contributes a additional space char. Oversensitive builder scripts are a problem in general, but here it seemed especially desirable to be able to change for example the default configuration for GHCJS packages without having to rebuild the whole GHC package sets, too, for really no good reason.

Now that I think again about it, it would probably be less invasive and more robust, to use a different function than `concatStringsSep` to achieve the same effect.